content/projects/gsoc/gsoc2018-project-ideas.adoc

```diff
@@ -0,0 +1,157 @@
+
+=== Role Strategy Plugin performance and/or user experience
+
+link:https://wiki.jenkins.io/display/JENKINS/Role+Strategy+Plugin[Role Strategy Plugin] is widely used in Jenkins as an authorization engine,
+but it has known performance limitations for large-scale setups.
+In this project the proposal is to improve the plugin's performance and to create a proper testing framework for Jenkins security plugins.
+Web interfaces of this plugin can be also improved.
+
+Skills:
+
+* Java
+* Performance testing and profiling tools
+* For UX projects - Javascript is a plus
+
+Potential mentors:
+link:https://github.com/oleg-nenashev/[Oleg Nenashev],
+link:https://github.com/christ66[Steven Christou]
+
+
+=== Integration plugin(s) for Electronic Design Automation tools
+
+The idea is to create a Jenkins plugin for one of widely used EDA tools.
+Both ASIC or FPGA design flow are acceptable, the tool should be proposed by the potential student.
+Open-source EDA tools would be preferable (e.g. Yosys, FuseSoC, ArachnePnR, icetools), but we also consider
+conditionally-free tools (like FPGA design EDAs).
+
+For example such plugin could report FPGA resource utilization per build as well as the project trend.
+Other example: timing report trend.
+
+Integrating UVM reports into Jenkins build and project pages is also an example of such plugin.
+
+Prerequisites:
+
+* Basic knowledge of Jenkins (as a user)
+* Basic knowledge of Java programming language
+* Hands-on experience with the selected EDA tool.
+In the case of FPGA flows it would be useful to have a prototyping board as well.
+
+Potential mentors:
+link:https://github.com/martinda[Martin d'Anjou],
+link:https://github.com/oleg-nenashev/[Oleg Nenashev]
+
+=== Code Coverage API Plugin
+
+There are a lot of plugins which currently implement code coverage (clover plugin, cobertura plugin, clover php plugin, etc),
+however they all are using similar charts and similar content.
+Instead of having one plugin for each code coverage result it would be best if we could merge them all into one plugin which will handle code coverage.
+
+Most code coverage results display similar content like:
+
+* File Name
+* Lines covered
+* Methods covered
+* Branches covered
+* Red/Green/Yellow statuses
+* Coverage over multiple builds
+
+Skills:
+
+* Java
+* HTML/CSS (possibly)
+
+Potential mentors:
+link:https://github.com/christ66[Steven Christou]
+
+=== Improvements to the Jenkins Acceptance Test Harness
+
+The link:https://github.com/jenkinsci/acceptance-test-harness[Jenkins Acceptance Test Harness (ATH)] is a great vehicle
+to test Jenkinsfiles and custom DSL libraries ahead of deploying them to production servers.
+However, it has couple of drawbacks.
+
+* it can be quite slow as it needs to bootstrap an entire Jenkins instance for each test method.
+* real production environments typically need to use a very specific plugin list of pre-defined plugins and plugin versions
```

Comment:
"very specific list of pre-defined plugins and their versions" would be a little more concise.